plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/application/ResourceLabelsPage.java

```diff
@@ -0,0 +1,233 @@
+		GridLayoutFactory.fillDefaults().margins(10, 10).applyTo(container);
+
+		Group labelsGroup = new Group(container, SWT.NONE);
+		labelsGroup.setText("Labels");
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).grab(true, true).applyTo(labelsGroup);
+		GridLayoutFactory.fillDefaults()
+				.numColumns(2).margins(6, 6).applyTo(labelsGroup);
+		Composite tableContainer = new Composite(labelsGroup, SWT.NONE);
+
+		this.viewer = createTable(tableContainer);
+		GridDataFactory.fillDefaults()
+				.span(1, 5).align(SWT.FILL, SWT.FILL).grab(true, true).applyTo(tableContainer);
+		ValueBindingBuilder.bind(ViewerProperties.singleSelection().observe(viewer))
+				.to(BeanProperties.value(IResourceLabelsPageModel.PROPERTY_SELECTED_LABEL).observe(model))
+				.in(dbc);
+		viewer.setContentProvider(new ObservableListContentProvider());
+		viewer.setInput(BeanProperties.list(
+				IResourceLabelsPageModel.PROPERTY_LABELS).observe(model));
+
+		Button addButton = new Button(labelsGroup, SWT.PUSH);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).applyTo(addButton);
+		addButton.setText("Add");
+		addButton.addSelectionListener(onAdd());
+		
+		Button editExistingButton = new Button(labelsGroup, SWT.PUSH);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).applyTo(editExistingButton);
+		editExistingButton.setText("Edit");
+		editExistingButton.addSelectionListener(onEdit());
+		ValueBindingBuilder
+				.bind(WidgetProperties.enabled().observe(editExistingButton))
+				.notUpdatingParticipant()
+				.to(BeanProperties.value(IResourceLabelsPageModel.PROPERTY_SELECTED_LABEL).observe(model))
+				.converting(new IsNotNullOrReadOnlyBooleanConverter())
+				.in(dbc);
+		
+		Button removeButton = new Button(labelsGroup, SWT.PUSH);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).applyTo(removeButton);
+		removeButton.setText("Remove");
+		removeButton.addSelectionListener(onRemove());
+		ValueBindingBuilder
+				.bind(WidgetProperties.enabled().observe(removeButton))
+				.notUpdatingParticipant()
+				.to(BeanProperties.value(IResourceLabelsPageModel.PROPERTY_SELECTED_LABEL).observe(model))
+				.converting(new IsNotNullOrReadOnlyBooleanConverter())
+				.in(dbc);
+	}
+
+	private Label getSelectedEntry() {
+		return (Label) viewer.getStructuredSelection().getFirstElement();
+	}
+	
+	private SelectionListener onRemove() {
+		return new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				Label label = getSelectedEntry();
+				if(MessageDialog.openQuestion(getShell(), "Remove Label", NLS.bind("Are you sure you want to delete the label {0} ", label.getName()))) {
+					model.removeLabel(label);
+					viewer.refresh();
+				}
+			}
+			
+		};
+	}
+
+	private SelectionListener onEdit() {
+		return new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				final Label label = getSelectedEntry();
```

Comment:
dont you want to get this from the model?
Or if not then why not using the util method in org.jboss.tools.openshift.internal.common.ui.utils.UIUtils.getFirstElement(selection)?